protocol/messenger_handler.go

```diff
@@ -320,29 +319,42 @@ func (m *Messenger) createContactRequestNotification(contact *Contact, messageSt
 			}
 		}
 
-		contactRequest = &common.Message{}
-
-		contactRequest.WhisperTimestamp = messageState.CurrentMessageState.WhisperTimestamp
-		contactRequest.Seen = true
-		contactRequest.Text = "Please add me to your contacts"
-		contactRequest.From = contact.ID
-		contactRequest.ContentType = protobuf.ChatMessage_CONTACT_REQUEST
-		contactRequest.Clock = messageState.CurrentMessageState.Message.Clock
-		contactRequest.ID = defaultID
-		contactRequest.ContactRequestState = common.ContactRequestStatePending
-		err = contactRequest.PrepareContent(common.PubkeyToHex(&m.identity.PublicKey))
+		// generate request message
+		contactRequest, err = m.generateContactRequest(
+			messageState.CurrentMessageState.Message.Clock,
+			messageState.CurrentMessageState.WhisperTimestamp,
+			contact,
+		)
 		if err != nil {
 			return err
 		}
 
+		// save this message
 		messageState.Response.AddMessage(contactRequest)
-
 		err = m.persistence.SaveMessages([]*common.Message{contactRequest})
 		if err != nil {
 			return err
 		}
 	}
 
+	if contactRequest != nil && contactRequest.ContactRequestState == common.ContactRequestStateAccepted {
+		// Pull one from the db if there
+		notification, err := m.persistence.GetActivityCenterNotificationByID(types.FromHex(contactRequest.ID))
+		if err != nil {
+			return err
+		}
+		if notification != nil {
+			err := m.persistence.UpdateActivityCenterNotificationMessage(notification.ID, contactRequest)
```

Comment:
FYI: In [this](https://github.com/status-im/status-go/pull/3126) PR i'm going to replace `UpdateActivityCenterNotificationMessage` calls with `SaveActivityCenterNotification `